homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -168,79 +168,136 @@ def supported_features(self):
             return SUPPORT_VIERATV | SUPPORT_TURN_ON
         return SUPPORT_VIERATV
 
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "name": self.name,
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "manufacturer": DEVICE_MANUFACTURER,
+        }
+
+    def update(self):
+        """Retrieve the latest data and renew connection."""
+        if self._remote:
+            try:
+                self._muted = self._remote.get_mute()
+                self._volume = self._remote.get_volume() / 100
+                self._state = STATE_ON
+            except OSError:
+                self._state = STATE_OFF
+
+        old_remote = self._remote
+        try:
+            if self._app_id and self._encryption_key:
+                self._remote = RemoteControl(
+                    self._host,
+                    self._port,
+                    app_id=self._app_id,
+                    encryption_key=self._encryption_key,
+                )
+            else:
+                self._remote = RemoteControl(self._host, self._port)
+        except Exception as err:
+            if old_remote:
+                _LOGGER.error("Could not establish remote connection: " + repr(err))
+
+    def send_key(self, key):
+        """Send a key to the tv and handles exceptions."""
+        if self._remote:
+            try:
+                self._remote.send_key(key)
+                self._state = STATE_ON
+            except OSError:
+                self._state = STATE_OFF
+                return False
```

Comment:
Nothing is check the return value of `send_key`.
```suggestion
                return
```